weave/utils/iterators.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+from collections.abc import Generator, Iterable, Iterator, Sequence
+from threading import Lock
+from typing import Any, TypeVar, overload
+
+T = TypeVar("T")
+
+
+class ThreadSafeInMemoryIteratorAsSequence(Sequence[T], Iterable[T]):
```

Comment:
All sequences are iterable, but not all iterables are sequences.

Is this supposed to represent an iterable, or a sequence?